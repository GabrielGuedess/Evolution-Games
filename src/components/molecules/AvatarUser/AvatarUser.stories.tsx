import { Story, Meta } from '@storybook/react/types-6-0';

import { AvatarUser } from './AvatarUser';

export default {
  title: 'Molecules/AvatarUser',
  component: AvatarUser,
} as Meta;

export const Default: Story = () => <AvatarUser />;
