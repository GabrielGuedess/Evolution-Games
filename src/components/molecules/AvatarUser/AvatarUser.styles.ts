import * as Avatar from '@radix-ui/react-avatar';
import * as Popover from '@radix-ui/react-popover';

import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  width: 3rem;
  height: 3.3rem;
`;

export const AvatarImage = styled(Avatar.Image)`
  width: 3rem;
  object-fit: cover;
  aspect-ratio: 1/1;
  border-radius: 50%;
`;

export const Trigger = styled(Popover.Trigger)`
  cursor: pointer;
`;

export const UserLink = styled.a`
  ${({ theme }) => css`
    cursor: pointer;
    color: ${theme.colors.whiteText};
    transition: color 0.3s linear;

    :is(:hover, :focus) {
      color: ${theme.colors.primary};
    }
  `}
`;
