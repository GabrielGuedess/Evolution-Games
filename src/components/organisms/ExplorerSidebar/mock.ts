const filterCategories = {
  title: 'Genre',
  name: 'genre',
  type: 'checkbox',
  fields: [
    { title: 'Action-adventure', name: 'action-adventure' },
    { title: 'Adventure', name: 'adventure' },
    { title: 'Sports', name: 'sports' },
    { title: 'Puzzle', name: 'puzzle' },
    { title: 'Horror', name: 'horror' },
    { title: 'Platform', name: 'platform' },
    { title: 'Fantasy', name: 'fantasy' },
    { title: 'RPG', name: 'role-playing' },
    { title: 'JRPG', name: 'jrpg' },
    { title: 'Simulation', name: 'simulation' },
    { title: 'Strategy', name: 'strategy' },
    { title: 'Shooter', name: 'shooter' },
  ],
};

const filterYear = {
  title: 'Year',
  name: 'year',
  type: 'checkbox',
  fields: [
    { title: '1990 - 2000', name: '1990-2000' },
    { title: '2001 - 2005', name: '2001-2005' },
    { title: '2006 - 2011', name: '2006-2011' },
    { title: '2012 - 2017', name: '2012-2017' },
    { title: '2018 - Now', name: '2018-now' },
  ],
};

const filterPlatforms = {
  title: 'Platforms',
  name: 'platforms',
  type: 'checkbox',
  fields: [
    {
      title: 'Playstation 4',
      name: 'ps4',
    },
    {
      title: 'Playstation 5',
      name: 'ps5',
    },
    {
      title: 'Xbox One',
      name: 'one',
    },
    {
      title: 'Xbox Series X',
      name: 'xs',
    },
  ],
};

const filterFeedback = {
  title: 'Notas',
  name: 'feedback',
  type: 'checkbox',
  fields: [
    { title: 'From 5.0', name: 'from-5.0' },
    { title: 'From 4.0', name: 'from-4.0' },
  ],
};

export const filterItems = [
  filterCategories,
  filterYear,
  filterPlatforms,
  filterFeedback,
];
