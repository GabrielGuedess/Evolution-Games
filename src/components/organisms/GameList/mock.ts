import { GameCardProps } from 'components/molecules/GameCard/GameCard';

const gameCardItems: GameCardProps[] = [
  {
    id: '1',
    title: 'God of War: Ragnar√∂k',
    genre: 'Action-adventure ',
    developer: 'Santa Monica Games',
    year: '2022',
    img: 'https://i.pinimg.com/originals/34/f8/e5/34f8e55845212171e8bb35edc5b08018.jpg',
    price: '$ 60.99',
    score: 5.0,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#9db4d0',
  },
  {
    id: '2',
    title: 'Star Wars Jedi: Fallen Order',
    genre: 'Action-adventure ',
    developer: 'EA',
    year: '2019',
    img: 'https://cdn2.steamgriddb.com/file/sgdb-cdn/grid/935a281a70ae21d54479c9c468145f7e.png',
    price: '$ 30.00',
    score: 3.8,
    platform: ['playstation', 'pc', 'xbox'],
    slug: '',
    primaryColor: '#a8b6cb',
  },
  {
    id: '3',
    title: 'Cyberpunk 2077',
    genre: 'Action-adventure ',
    developer: 'CD PROJEKT RED',
    year: '2020',
    img: 'https://whatifgaming.com/wp-content/uploads/2022/05/Neon-Aesthetic.jpg',
    price: '$ 10.00',
    score: 1.8,
    platform: ['playstation', 'pc', 'xbox'],
    slug: '',
    primaryColor: '#dabbde',
  },
  {
    id: '4',
    title: 'The Last of Us Part II',
    genre: 'Action-adventure ',
    developer: 'Naughty Dog',
    year: '2020',
    img: 'https://www.jeuxvideo-live.com/wp-content/uploads/jvl/2022/07/le-multi-de-the-last-of-us-2-deevoile-quelques-secrets-de-conception-104207-large.jpg',
    price: '$ 60.99',
    score: 5.0,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#b2a7a5',
  },
  {
    id: '5',
    title: 'Elden Ring',
    genre: 'Action-adventure ',
    developer: 'BANDAI NAMCO Entertainment Inc',
    year: '2022',
    img: 'https://image.api.playstation.com/vulcan/ap/rnd/202110/2000/aGhopp3MHppi7kooGE2Dtt8C.png',
    price: '$ 60.99',
    score: 4.8,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#aeb195',
  },
  {
    id: '6',
    title: 'Horizon Forbidden West',
    genre: 'Action-adventure ',
    developer: 'Santa Monica Games',
    year: '2022',
    img: 'https://t.ctcdn.com.br/efyy2ADqd67kPvUfXBVJWMJiX_I=/1400x788/smart/filters:format(webp)/i499600.png',
    price: '$ 60.99',
    score: 4.8,
    platform: ['playstation'],
    slug: '',
    primaryColor: '#aeb1d2',
  },
  {
    id: '7',
    title: "Marvel's Spider-Man",
    genre: 'Action-adventure ',
    developer: 'Insomniac Games',
    year: '2018',
    img: 'https://image.api.playstation.com/vulcan/ap/rnd/202008/1020/b0OfuUs4lHGWG4VNlHcXwL5a.jpg',
    price: '$ 60.99',
    score: 4.7,
    platform: ['playstation', 'pc'],
    slug: '',
    primaryColor: '#9ab3c5',
  },
  {
    id: '8',
    title: 'The Witcher 3',
    genre: 'Action-adventure ',
    developer: 'CD PROJEKT SA',
    year: '2015',
    img: 'https://e.snmc.io/lk/f/x/1508cc32995d91b6206ee1b1e82b9338/8329790',
    price: '$ 60.99',
    score: 4.5,
    platform: ['playstation', 'pc', 'xbox'],
    slug: '',
    primaryColor: '#d1b2a3',
  },
];

export default gameCardItems;
