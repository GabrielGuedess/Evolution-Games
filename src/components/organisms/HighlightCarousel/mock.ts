import { HighlightGameProps } from 'components/molecules/HighlightGame/HighlightGame';

export const highlightMock: HighlightGameProps[] = [
  {
    img: '/img/games/starWarsHighlight.png',
    gameLogo: '/img/games/highlightName/nameStarWars.png',
    gameLogoAlt: 'Logo Star Wars',
    alt: 'Star Wars',
    releaseYear: new Date(2019, 10, 10),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['pc', 'playstation', 'xbox'],
  },
  {
    img: '/img/games/r8.jpg',
    gameLogo: '/img/games/highlightName/nameR8.png',
    gameLogoAlt: 'Logo Star Wars',
    alt: 'Resident Evil 8',
    releaseYear: new Date('09-10-2022'),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['pc', 'playstation', 'xbox'],
  },
  {
    img: '/img/games/starWars.jpg',
    gameLogo: '/img/games/highlightName/nameStarWars.png',
    gameLogoAlt: 'Logo Star Wars 2',
    alt: 'Star Wars 2',
    releaseYear: new Date('09-10-2022'),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['pc', 'playstation', 'xbox'],
  },
  {
    img: '/img/games/theLastOfUs.jpg',
    gameLogo: '/img/games/highlightName/nameTheLastOfUs.png',
    gameLogoAlt: 'Logo The last Of Us',
    alt: 'The last Of Us',
    releaseYear: new Date('09-10-2022'),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['playstation'],
  },
];
