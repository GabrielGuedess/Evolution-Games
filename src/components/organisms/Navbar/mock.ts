import { GameCartDropProps } from 'components/molecules/GameCartDrop/GameCartDrop';

export const cartItems: GameCartDropProps[] = [
  {
    id: '1',
    name: 'God of War: Ragnar√∂k',
    genres: [{ name: 'Action-adventure ', id: '1', slug: 'action-adventure' }],
    developers: [
      { id: '1', name: 'Santa Monica Games', slug: 'santa-monica-games' },
    ],
    background:
      'https://i.pinimg.com/originals/34/f8/e5/34f8e55845212171e8bb35edc5b08018.jpg',
    price: 250.0,
    platforms: [{ name: 'Plastation 5', slug: 'ps5', id: '1' }],
    quantity: 1,
  },
];
