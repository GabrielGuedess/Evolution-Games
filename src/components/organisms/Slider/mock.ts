import { SlideProps } from 'components/molecules/Slide/Slide';

export const SliderMock: SlideProps[] = [
  {
    srcSlideImage: '/img/games/theLastOfUs.png',
    altSlideImage: 'The Last Of Us',
    srcHighlightImage: '/img/games/highlightName/nameTheLastOfUs.png',
    altHighlightImage: 'The Last Of Us Highlight',
  },
  {
    srcSlideImage: '/img/games/god.jpg',
    altSlideImage: 'God Of War',
    srcHighlightImage: '/img/games/highlightName/nameGodOfWar.png',
    altHighlightImage: 'God Of War Highlight',
  },
  {
    srcSlideImage: '/img/games/r8.jpg',
    altSlideImage: 'Resident Evil Village',
    srcHighlightImage: '/img/games/highlightName/nameR8.png',
    altHighlightImage: 'Resident Evil Village Highlight',
  },
  {
    srcSlideImage: '/img/games/starWars.jpg',
    altSlideImage: 'Star Wars',
    srcHighlightImage: '/img/games/highlightName/nameStarWars.png',
    altHighlightImage: 'Star Wars Highlight',
  },
];
