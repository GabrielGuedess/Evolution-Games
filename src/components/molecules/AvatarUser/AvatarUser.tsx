import { useSession } from 'next-auth/react';
import Link from 'next/link';
import { useRouter } from 'next/router';

import * as Avatar from '@radix-ui/react-avatar';
import * as Popover from '@radix-ui/react-popover';
import Avvvatars from 'avvvatars-react';
import { User } from 'phosphor-react';

import { UserInfo } from 'components/molecules/UserInfo/UserInfo';

import * as S from './AvatarUser.styles';

export const AvatarUser = () => {
  const { data: session, status } = useSession();
  const { asPath } = useRouter();

  return (
    <S.Wrapper>
      <Popover.Root>
        <S.Trigger aria-label="User Photo Button">
          {status === 'authenticated' ? (
            <Avatar.Root>
              {session.user.photo ? (
                <S.AvatarImage
                  src={session.user.photo!}
                  alt="Image User"
                  aria-label="User Photo"
                />
              ) : (
                <Avvvatars size={30} value={session.user.email} />
              )}
              <Avatar.Fallback />
            </Avatar.Root>
          ) : (
            <Link href={`/sign-in?callbackUrl=${asPath}`} passHref>
              <S.UserLink aria-label="Link for Login">
                <User size={24} />
              </S.UserLink>
            </Link>
          )}
        </S.Trigger>
        <Popover.Anchor />
        <Popover.Content>
          {status === 'authenticated' && (
            <UserInfo
              userPhoto={session.user.photo}
              username={session.user.username}
              email={session.user.email}
              name={session.user.name}
            />
          )}

          <Popover.Arrow />
        </Popover.Content>
      </Popover.Root>
    </S.Wrapper>
  );
};
