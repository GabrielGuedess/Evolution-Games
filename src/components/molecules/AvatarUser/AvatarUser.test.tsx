import { renderWithTheme } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { userInfo } from 'templates/Home/mock';

import { AvatarUser } from './AvatarUser';

describe('<AvatarUser />', () => {
  it('should render the photo user', () => {
    // Arrange
    const { container } = renderWithTheme(<AvatarUser {...userInfo} />);

    // Assert
    expect(screen.getByLabelText('User Photo Button')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render without the photo user', () => {
    renderWithTheme(<AvatarUser />);

    // Assert
    expect(screen.getByLabelText('Link for Login')).toBeInTheDocument();
  });
});
