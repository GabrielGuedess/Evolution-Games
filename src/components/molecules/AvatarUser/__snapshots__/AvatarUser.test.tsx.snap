// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<AvatarUser /> should render the photo user 1'] = `
.c0 {
  cursor: pointer;
}

<div
  class=""
>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    aria-label="User Photo Button"
    class="c0"
    data-state="closed"
    type="button"
  >
    <span>
      <span />
    </span>
  </button>
  <div />
</div>
`;
